shunit2

```diff
@@ -630,7 +630,12 @@ failNotEquals() {
   shunit_actual_=$2
 
   shunit_message_=${shunit_message_%% }
-  _shunit_assertFail "${shunit_message_:+${shunit_message_} }expected:<${shunit_expected_}> but was:<${shunit_actual_}>"
+  if command [ $(echo "${shunit_expected_}" | wc -l) -gt 10 ] || command [ $(echo "${shunit_actual_}" | wc -l) -gt 10 ]; then
+      diff_output_=$(diff <(echo "${shunit_expected_}") <(echo "${shunit_actual_}"))
```

Comment:
I recognize process substitution `<()` as a Bashism, though I don't know the details.

Reply:
This is process substitution, it provides a way of passing the output of a command to another command where pipe is not available. diff takes filenames, but the items that want to be diff'd are in variables.
I suspect this might be why the tests are failing, as it probably doesn't work in other shells.
If anyone has any suggestions on how I can do this differently then let me know.